### DISCUSSION TITLE
Error logging into OneDrive

### DISCUSSION THREAD

**Discussion by brauliodiasribeiro** [abraunegg/onedrive#2357] [Q&A] [answered]
Anyone with this problem?
```
docker start 7927a5e88cf2 --interactive
usermod: no changes
Base Args: --monitor
# Launching onedrive
Cannot connect to Microsoft OneDrive Service - Network Connection Issue

ERROR: Microsoft OneDrive API returned an error with the following message:
Error Message:    Timeout was reached on handle 55E8409B16F0
Unable to reach Microsoft OneDrive API service at this point in time, re-trying network tests

Cannot connect to Microsoft OneDrive Service - Network Connection Issue
```

I tested this commands and the results it´s OK.
```
root@SERVER:~# curl -I --connect-timeout 5 --max-time 5 https://login.microsoftonline.com
curl: (28) Resolving timed out after 5001 milliseconds
root@SERVER:~# curl -I --connect-timeout 5 --max-time 5 https://graph.microsoft.com
curl: (28) Resolving timed out after 5001 milliseconds}
```

**Comment by abraunegg** [comment, marked as answer]
@bdrtec 
Your tests/results are *not* OK - they *fail* with a timeout.

You essentially have poor DNS resolution. Please read  
https://github.com/abraunegg/onedrive/wiki/Frequently-Asked-Questions#cannot-connect-to-microsoft-onedrive-service---network-connection-issue carefully.


Your options to solve are:
* Use better DNS Servers. Speak to your ISP.
* Install and maintain a DNS server on your local LAN to cache your DNS queries
* Update your `/etc/resolv.conf` file as you may have a broken firewall or router that is messing with your DNS responses.
